evaluation/datasets/common.py

```diff
@@ -285,11 +286,14 @@ def eval(self, predictions: list, targets=None, perturb_prefix=None) -> dict:
                         k: get_eval_metrics(self.task, *list(zip(*examples)))
                         for k, examples in examples_by_tag.items()
                     }
-                    score_obj["metadata_json"]["perf_by_tag"] = [
+                    score_obj["metadata_json"]["perf_by_tag"] = score_obj[
+                        "metadata_json"
```

Comment:
Yeah, the evaluation code is starting to get a little messy. Now, perf_by_tag can also come directly from a metric. This is useful to have, because we want the dataperf_accuracy metric to return performances for each label in the dataset individually.

Reply:
It could be good to make an issue for rethinking the evaluation architecture, and the scores table.